packages/unlock-js/src/PublicLock/utils/getPurchaseKeysArguments.js

```diff
@@ -0,0 +1,54 @@
+import formatKeyPrice from './formatKeyPrice'
+import utils from '../../utils'
+
+export default async function getPurchaseKeysArguments({
+  lockAddress,
+  erc20Address,
+  decimals,
+  owners = [],
+  keyManagers = [],
+  keyPrices = [],
+  referrers = [],
+  data = [],
+}) {
+  // If erc20Address was not provided, get it
+  const lockContract = await this.getLockContract(lockAddress)
+  erc20Address = await lockContract.tokenAddress()
```

Comment:
shoould we do that only if `erc20Address` is not set?

Reply:
It's just getting an erc20address from lock if none provided. It's to avoid overfetching if user already got erc20 for lock.